chameleon_mailman/tasks.py

```diff
@@ -1,10 +1,38 @@
 from __future__ import absolute_import, unicode_literals
 
-from celery import shared_task
+from datetime import datetime, timedelta
+import logging
+import pytz
 
+from celery import task
+
+from django.conf import settings
 from django.core.mail import send_mail
+from django.utils.html import strip_tags
+from user_news.models import Outage
+
+logger = logging.getLogger(__name__)
+
+
+@task
+def send_outage_reminders(crontab_frequency, send_outage_reminder_before):
+    now = pytz.utc.localize(datetime.utcnow())
+    upcoming_outages = Outage.objects.filter(start_date__gte=now)
+
+    send_window_max = now + timedelta(seconds=send_outage_reminder_before)
+    send_window_min = send_window_max - timedelta(
+        seconds=(crontab_frequency * 60))
+
+    for outage in upcoming_outages.all():
+        if send_window_min < outage.start_date < send_window_max:
```

Comment:
The frequency is 30 minutes, but the total window is 2 hours. Unless we somehow store that we've already sent mail, this will send 4 mails per outage, no? That's why I was advocating for adding a model field to store something like `notification_sent_at`, storing a date of when the notification was sent (better for auditing than a simple boolean `notification_sent`, which only tells you if it was sent or not).

Reply:
You could test this by setting interval very low, like 1 minute, and the "remind before" window to 5 minutes. I'd expect 5 mails to be generated from the test environment.